Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2055,7 +2071,10 @@ namespace Babylon
 
             static Napi::Object New(const Napi::CallbackInfo& info)
             {
-                return info.Env().Global().Get(JS_CLASS_NAME).As<Napi::Function>().New({});
+                // Instead of creating just a C++ XRFrame object, wrap it in a JS NativeXRFrame object (see nativeXRFrame.ts in Babylon.JS)
+                const auto nativeImpl = info.Env().Global().Get(JS_CLASS_NAME).As<Napi::Function>().New({});
+                const auto nativeXRFrame = JsRuntime::NativeObject::GetFromJavaScript(info.Env()).Get("NativeXRFrame").As<Napi::Function>().New({nativeImpl}).As<Napi::Object>();
```

Comment:
Is this called often? Might consider caching the NativeXRFrame constructor somehow.

Reply:
It's actually called only once when the session is created, and then the single frame instance is updated every frame with new frame data from the XR runtime